// @azure/cognitiveservices-qnamaker/esm/operations/endpointKeys.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/endpointKeysMappers";
import * as Parameters from "../models/parameters";
/** Class representing a EndpointKeys. */
var EndpointKeys = /** @class */ (function () {
    /**
     * Create a EndpointKeys.
     * @param {QnAMakerClientContext} client Reference to the service client.
     */
    function EndpointKeys(client) {
        this.client = client;
    }
    EndpointKeys.prototype.getKeys = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, getKeysOperationSpec, callback);
    };
    EndpointKeys.prototype.refreshKeys = function (keyType, options, callback) {
        return this.client.sendOperationRequest({
            keyType: keyType,
            options: options
        }, refreshKeysOperationSpec, callback);
    };
    return EndpointKeys;
}());
export { EndpointKeys };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getKeysOperationSpec = {
    httpMethod: "GET",
    path: "endpointkeys",
    urlParameters: [
        Parameters.endpoint
    ],
    responses: {
        200: {
            bodyMapper: Mappers.EndpointKeysDTO
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
var refreshKeysOperationSpec = {
    httpMethod: "PATCH",
    path: "endpointkeys/{keyType}",
    urlParameters: [
        Parameters.endpoint,
        Parameters.keyType
    ],
    responses: {
        200: {
            bodyMapper: Mappers.EndpointKeysDTO
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    serializer: serializer
};
