// @azure/cognitiveservices-qnamaker/esm/operations/operations.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/operationsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Operations. */
var Operations = /** @class */ (function () {
    /**
     * Create a Operations.
     * @param {QnAMakerClientContext} client Reference to the service client.
     */
    function Operations(client) {
        this.client = client;
    }
    Operations.prototype.getDetails = function (operationId, options, callback) {
        return this.client.sendOperationRequest({
            operationId: operationId,
            options: options
        }, getDetailsOperationSpec, callback);
    };
    return Operations;
}());
export { Operations };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getDetailsOperationSpec = {
    httpMethod: "GET",
    path: "operations/{operationId}",
    urlParameters: [
        Parameters.endpoint,
        Parameters.operationId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.Operation,
            headersMapper: Mappers.OperationsGetDetailsHeaders
        },
        default: {
            bodyMapper: Mappers.ErrorResponse,
            headersMapper: Mappers.OperationsGetDetailsHeaders
        }
    },
    serializer: serializer
};
